src/app/users/users.component.html

```diff
@@ -1,21 +1,56 @@
-<h1 i18n>Users</h1>
-<div class="km-user-table" *ngIf="displayTable">
-  <form (ngSubmit)="roleSubmit(allUsers.data,selectedRole)" #rolesForm="ngForm">
-    <mat-form-field>
-      <mat-select [(ngModel)]="selectedRole" name="role">
-        <mat-option *ngFor="let role of roleList" [value]="role">{{role}}</mat-option>
-      </mat-select>
-    </mat-form-field>
-    <button mat-button type="submit" i18n>Add role to selected</button>
-  </form>
-  <mat-table #table [dataSource]="allUsers">
+<mat-toolbar>
+  <button mat-icon-button (click)="back()"><mat-icon>arrow_back</mat-icon></button>
+  <span i18n>Users</span>
+  <span class="toolbar-fill"></span>
+  <mat-icon>search</mat-icon>
+  <mat-form-field class="font-size-1">
+    <input matInput (keyup)="applyResFilter($event.target.value)" i18n-placeholder placeholder="Type text to search...">
+  </mat-form-field>
+</mat-toolbar>
+
+<div class="space-container" *ngIf="displayTable">
+  <mat-toolbar class="primary-color font-size-1">
+    <span class="toolbar-fill">
+      <h1 i18n>All Users</h1>
+    </span>
+    <div class="margin-lr-3">
+      <form (ngSubmit)="roleSubmit(allUsers.data,selectedRole)" #rolesForm="ngForm">
+        <mat-form-field>
+          <mat-select [(ngModel)]="selectedRole" name="role">
+            <mat-option *ngFor="let role of roleList" [value]="role">{{role}}</mat-option>
+          </mat-select>
+        </mat-form-field>
+        <button mat-button type="submit"><mat-icon>library_add</mat-icon><span i18n>Add Roles</span></button>
+      </form>
+    </div>
+    <mat-paginator #paginator [pageSize]="10" [pageSizeOptions]="[5, 10, 20]" class="primary-color"></mat-paginator>
+  </mat-toolbar>
+  <mat-table #table [dataSource]="allUsers" matSort>
+    <ng-container matColumnDef="select">
+      <mat-header-cell *matHeaderCellDef>
+        <mat-checkbox (change)="$event ? masterToggle() : null"
+        [checked]="selection.hasValue() && isAllSelected()"
+        [indeterminate]="selection.hasValue() && !isAllSelected()" i18n>
+      </mat-checkbox>
+      </mat-header-cell>
+      <mat-cell *matCellDef="let row">
+        <mat-checkbox (click)="$event.stopPropagation()"
+          (change)="$event ? selection.toggle(row) : null"
+          [checked]="selection.isSelected(row)">
+        </mat-checkbox>
+      </mat-cell>
+    </ng-container>
+    <ng-container matColumnDef="profile">
+      <mat-header-cell *matHeaderCellDef i18n>Profile Image</mat-header-cell>
+      <mat-cell *matCellDef="let element" (click)="select(element)">{{element._attachments}}</mat-cell>

```

Comment:
This should be an img with the profile picture as src.  We should consider renaming the profile image to always have the same name which will make setting the src easier.